api/content-script/listeners/wallet-settings-updated.ts

```diff
@@ -0,0 +1,18 @@
+import { ApiUtils } from "@/api/background/utils.ts";
+import {
+  WALLET_SETTINGS,
+  WalletSettings,
+} from "@/contexts/WalletManagerContext.tsx";
+
+export const watchWalletSettingsUpdated = () => {
+  storage.watch<WalletSettings>(WALLET_SETTINGS, async (newValue, oldValue) => {
+    if (
+      newValue?.selectedWalletId !== oldValue?.selectedWalletId ||
+      newValue?.selectedAccountIndex !== oldValue?.selectedAccountIndex
+    ) {
+      const account = await ApiUtils.getCurrentAccount();
+
+      ApiUtils.createApiResponse("kas:account_changed", account?.address);
```

Comment:
```suggestion
      ```suggestion
     const isConnected = await ApiUtils.isHostConnected(window.location.host);
      window.postMessage(
        ApiUtils.createApiResponse("kas:account_changed", isConnected ? [account?.address] ? []),
        window.location.origin,
      );
```
```